utils/httpUtils.ts

```diff
@@ -72,3 +71,114 @@ export const notFound = (message: string) => {
     message: message,
   });
 };
+
+export const getFeatureFlagsFromRequest = context => {
+  return context.request.headers.get("x-should-read-from-nextgen");
+};
+
+export const shouldReadFromNextGen = async context => {
+  let shouldReadFromNextGen = getFeatureFlagsFromRequest(context);
+  if (shouldReadFromNextGen === true || shouldReadFromNextGen === "true" || shouldReadFromNextGen === "True") {
+    // if the header is set, return the value
+    return true;
+  }
+  return false;
+};
+
+export const formatFedQueryForNextGen = (query: string) => {
+  let cleanQuery = query;
+  // remove fed Prefix
+  const firstCurlyBracket = query.indexOf("{");
+  const secondCurlyBracket = query.indexOf("{", firstCurlyBracket + 1);
+  const fedIndex = query.indexOf("fed");
+  if (fedIndex !== -1 && fedIndex < secondCurlyBracket && fedIndex > firstCurlyBracket) {
+    const splitOnFirstFed = [query.substring(0, fedIndex + 4), query.substring(fedIndex + 4)];
+    const splitQueryOnFed = splitOnFirstFed[0].split("fed");
+    const firstLetterLowerCase = splitQueryOnFed[1][0].toLowerCase();
+    splitQueryOnFed[1] = splitQueryOnFed[1].slice(1);
+    splitQueryOnFed.splice(1, 0, firstLetterLowerCase);
+    cleanQuery = splitQueryOnFed.join("").concat(splitOnFirstFed[1]);
+  }
+
+  // remove input object from variables
+  const finishedQuery = cleanQuery
+    .replace("input: {", "")
+    .replace("}}", "}")
+    // apply any specific type switches that need to be made - these can be passed in from the resolver
+    .replace("$workflowRunId: String", "$workflowRunId: UUID") // lets make an actual UUID
+    .replace(/query_fedConsensusGenomes_items/g, "ConsensusGenome");
```

Comment:
why does this one have `_items` and the others don't?

Reply:
oh yeah, is this consensus genome stuff a leftover hack from the demo?

Reply:
This is to make the Fragment part of the query work with next gen. The type that the info is on in Next Gen is different than the type that it is on in Fed server. It is a hack from the demo but I have no plans on getting rid of it.

Reply:
This can evolve if it needs to if we end up using this function regularly. Yeah?